File: codecheff/LTIME77B/4.cpp
#include <iostream>
#include <vector>
#include <cstring>
using namespace std;
int t, n, q, k, ans;
vector<int> numbers;
int invCount[32][2];

void solve(int bit, vector<int> v)
{
    if (bit == -1 || v.empty())
        return;
    vector<int> v1, v2;
    int z, o;
    z = o = 0;
    for (auto ele : v)
    {
        if (ele & (1 << bit))
        {
            v1.push_back(ele);
            o++;
            invCount[bit][1] += z;
        }
        else
        {
            v2.push_back(ele);
            z++;
            invCount[bit][0] += o;
        }
    }
    solve(bit - 1, v1);
    solve(bit - 1, v2);
}

int main()
{
    cin >> t;
    while (t--)
    {
        memset(invCount, 0, sizeof(invCount));
        cin >> n >> q;
        for (int i = 0; i < n; i++)
        {
            int temp;
            cin >> temp;
            numbers.push_back(temp);
        }
        solve(30, numbers);
        while (q--)
        {
            cin >> k;
            ans = 0;
            for (int bit = 0; bit < 31; bit++)
            {
                if (k & (1 << bit))
                    ans += invCount[bit][1];
                else
                    ans += invCount[bit][0];
            }
            cout << ans << endl;
        }
    }
}

File: codecheff/LTIME77B/4.py
t = int(input())
for _ in range(t):
    n, q = map(int, input().split())
    a = [int(x) for x in input().split()]
    for temp in range(q): 
        k = int(input())
        ans = 0
        b = a.copy()
        for i in range(n):
            b[i] = b[i] ^ k
        for i in range(n):
            for j in range(i + 1, n):
                if  b[j] < b[i]:
                   ans += 1
        print(ans)